addons/account/models/account_journal.py

```diff
@@ -604,6 +604,25 @@ def _alias_prepare_alias_name(self, alias_name, name, code, jtype, company):
                 alias_name = f"{alias_name}-{company_identifier}"
         return self.env['mail.alias']._sanitize_alias_name(alias_name)
 
+    @api.model
+    def _ensure_unique_alias_creation(self, vals, company):
+        """ Check uniqueness of the alias name within the given alias domain.
+        :param vals: the values of the journal.
+        :return: a unique alias name.
+        """
+        alias_name = vals['alias_name']
+        alias_domain_name = company.alias_domain_id.name
+
+        domain = [('alias_name', '=', alias_name)]
+        if alias_domain_name:
+            domain.append(('alias_domain', '=', alias_domain_name))
+        existing_alias = self.env['mail.alias'].search_count(domain, limit=1)
+
+        if existing_alias:
+            alias_name = f"{alias_name}-{vals.get('code', vals['name'][:5])}"
```

Comment:
Hello
Initially I wanted to use this same logic in case the code doesn't exist (here for imported journals) https://github.com/odoo/odoo/blob/6a497a26a3b1d00a08cb30e164a7d7e8499d6d61/addons/account/models/account_journal.py#L779

But looking at it again now that I understand things more, as @hupo-odoo mentioned there should be no case where the `code` doesn't exist at this point as the field is mandatory. It's safe to assume the code always exists here IMO?

But if not, I'm not sure how to make it more unique as numbers and random letters are not a good option :/

Reply:
I see, and the code will not try to fill the field `code` again after this point so putting an alternative value is useless (since creation is sure to fail). Thank you :)